pygsti/baseobjs/nicelyserializable.py

```diff
@@ -15,9 +15,10 @@
 import json as _json
 import numpy as _np
 import scipy.sparse as _sps
+from pygsti.baseobjs.mongoserializable import MongoSerializable
 
 
-class NicelySerializable(object):
+class NicelySerializable(MongoSerializable):
```

Comment:
Just to confirm the swap of inheritance order. The intention is that everything that is filesystem serializable is also mongo serializable, but there are some objects (e.g. DataSet,  Estimate, Protocol) which are currently only mongo serializable (as of this PR)? I assume the goal is to enable the storing/reuse of these subcomponents, which before were only dumped as part of a larger tree structure?

Reply:
Yes, this is correct, and I agree it's a bit counterintuitive at first.  The thinking is that "everything can be rendered as JSON can also be serialized to a MongoDB" which implies that anything that is `NicelySerializable` must also be `MongoSerializable`.  Another way of thinking about this is that something that is `NicelySerializable` is able to render itself as simple JSON, whereas something that is `MongoSerializable` is able to insert itself into a Mongo database, perhaps using multiple records.  In principle we could take all the records a `MongoSerializable` object generates and cram them together into one big piece of JSON, but in practice there are certain more complex objects (like `DataSet`s, `Estimate`s, and `Protocol`s) that naturally can save themselves as multiple records in a database but would be a bit unwieldy as a huge piece of JSON.